app/helpers/form_helper.rb

```diff
@@ -0,0 +1,33 @@
+module FormHelper
+  def form_errors(errors = [])
+    tag.ul(class: %w[help-block error-block]) do
+      safe_join(errors.map { |e| tag.li(e) }) if errors.present?
+    end
+  end
+
+  def form_group(form, field_name, label: nil, help: nil, attributes: {}, &block)
+    raise StandardError, "Form cannot be nil" if form.nil?
+    raise StandardError, "No input field given" unless block_given?
+
+    errors = form.object.errors[field_name]
+    attributes[:class] = Array(attributes[:class]) << "form-group"
+    attributes[:class] << "has-error" if errors && errors.any?
```

Comment:
Hi @kevindew - I've discovered an issue with this. There are 2 tests (e.g. [this one](https://github.com/alphagov/publisher/blob/3e928a81d339387f91699ba6fba8c4a4d1eb5231/test/functional/editions_controller_test.rb#L351)) that send a `Hash` through. They are not all `ActiveModel::Errors` objects as you would expect.

The original line 14 (above) sort of masked the issue and the tests passed. But I'm not 100% convinced it's the best approach. I've tried using the safe navigation operator `errors = form&.object&.errors[field_name] || []` which works, but needs the `|| []`, we could convert the tests to be instances of `ActiveModel::Errors`... But is there a better way?

Please let me know your preferred approach on this.  Thanks.

Reply:
Thanks,  I expect the problem in the example you linked to is this line:
https://github.com/alphagov/publisher/blob/3e928a81d339387f91699ba6fba8c4a4d1eb5231/test/functional/editions_controller_test.rb#L347

Where it mocks fake errors on the model

I'm thinking that rather than writing this code in order to satisfy mocking, we should make mocks more realistic. As an opening gambit I'm wondering if we can replace the stub of errors to be:

```
@guide.errors.add(:base, "Editions scheduled for publishing can't be edited")
```

Reply:
Yup, this works great - thanks 👍   Both instances updated.